src/helics/core/helicsVersion.cpp

```diff
@@ -0,0 +1,230 @@
+            osInfo.dwOSVersionInfoSize = sizeof(osInfo);
+            RtlGetVersion(&osInfo);
+            winVer +=
+                std::to_string(osInfo.dwMajorVersion) + '.' + std::to_string(osInfo.dwMinorVersion);
+        }
+    }
+    return winVer;
+}
+
+#else
+#    include <sys/utsname.h>
+#    include <unistd.h>
+
+std::uint64_t getTotalSystemMemory()
+{
+    std::uint64_t pages = sysconf(_SC_PHYS_PAGES);
+    std::uint64_t page_size = sysconf(_SC_PAGE_SIZE);
+    return pages * page_size;
+}
+
+std::string os_info()
+{
+    std::string osInfo;
+    struct utsname details = {};
+
+    int ret = uname(&details);
+
+    if (ret == 0) {
+        osInfo.append(details.sysname);
+        osInfo.push_back(' ');
+        osInfo.push_back(' ');
+        osInfo.append(details.release);
+        osInfo.push_back(' ');
+        osInfo.push_back(' ');
+        osInfo.append(details.version);
+    } else {
+        osInfo = "POSIX";
+    }
+    return osInfo;
+}
+#endif
+
+std::string getHostName()
+{
+    char* temp{nullptr};
+    std::string computerName;
+
+#if defined(WIN32) || defined(_WIN32) || defined(_WIN64)
+    temp = getenv("COMPUTERNAME");
+    if (temp != nullptr) {
+        computerName = temp;
+        temp = nullptr;
+    }
+#else
+    temp = getenv("HOSTNAME");
+    if (temp != nullptr) {
+        computerName = temp;
+        temp = nullptr;
+    } else {
+        temp = new char[512];
+        if (gethostname(temp, 512) == 0) {  // success = 0, failure = -1
+            computerName = temp;
+        }
+        delete[] temp;
+        temp = nullptr;
+    }
+#endif
+    return computerName;
+}
+
+namespace helics {
+
+std::string systemInfo()
+{
+    Json::Value base;
+    base["version"]["string"] = helics::versionString;
+    base["version"]["major"] = helics::versionMajor;
+    base["version"]["minor"] = helics::versionMinor;
+    base["version"]["patch"] = helics::versionPatch;
+    base["version"]["build"] = helics::versionBuild;
+    base["buildflags"] = helics::buildFlags;
+    base["compiler"] = helics::compiler;
+    base["cores"] = Json::arrayValue;
+
+    auto ctypesA = CoreFactory::getAvailableCoreTypes();
+    for (const auto& ctype : ctypesA) {
+        base["cores"].append(ctype);
+    }
+    auto cpumodel = getCPUModel();
+    if (!cpumodel.empty()) {
+        if (cpumodel.back() == ' ' || cpumodel.back() == '\n' || cpumodel.back() == '\0') {
+            cpumodel.pop_back();
+        }
+        base["cpu"] = cpumodel;
+    } else {
+        base["cpu"] = "UNKNOWN";
+    }
+    base["cpucount"] = std::thread::hardware_concurrency();
+    base["cputype"] = HELICS_BUILD_PROCESSOR;
+    base["hostname"] = getHostName();
+#if defined(HELICS_ENABLE_ZMQ_CORE) && !defined(USING_HELICS_C_SHARED_LIB)
+    base["zmqversion"] = helics::zeromq::getZMQVersion();
+#endif
+    auto memory = getTotalSystemMemory();
+    base["memory"] = std::to_string(memory / (1024ULL * 1024ULL)) + " MB";
+    base["OS"] = os_info();
```

Comment:
```suggestion
    base["os"] = os_info();
```
Matching all lowercase of other fields.